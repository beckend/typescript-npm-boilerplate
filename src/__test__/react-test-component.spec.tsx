/* tslint:disable: jsx-wrap-multiline */
/**
 * React test
 */
import * as React from 'react';
import {
  TestComponent,
  ITestComponentProps
} from '../react-test-component';
import {
  mount,
  ReactWrapper
} from 'enzyme';

const renderer = require('react-test-renderer');

describe('react component', () => {
  let wrapper: ReactWrapper<ITestComponentProps, {}>;

  beforeEach(() => jest.resetModules());

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
    }
  });

  it('Renders with enzyme', () => {
    wrapper = mount<ITestComponentProps, {}>(
      <TestComponent
        str1='My string'
      />
    );
    expect(wrapper.find(TestComponent).length)
      .toEqual(1);
  });

  it('Renders with jest test rendrer', () => {
    const component = renderer.create(
      <TestComponent
        str1='My string'
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
