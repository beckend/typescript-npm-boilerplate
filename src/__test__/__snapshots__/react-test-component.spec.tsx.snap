exports[`react component Renders with jest test rendrer 1`] = `
<div>
  My string
</div>
`;
